slurm_deploy/deploy_ctl.sh

```diff
@@ -273,302 +287,317 @@ function deploy_build_item() {
                 export PATH=$tools_path:$PATH && ./autogen.pl
             fi
         fi
+
+        # shellcheck disable=SC2181
         if [ "$?" != "0" ]; then
             echo_error $LINENO "\"$item\" Autogen error. Cannot continue."
-            rm configure 2> /dev/null
+            rm configure 2>/dev/null
             exit 1
         fi
     fi
+
     cd .build || (echo_error $LINENO "directory change error" && exit 1)
+
     if [ ! -f "config.log" ]; then
-        pdsh -S -w $build_node "cd $PWD && ./config.sh"
+        pdsh -S -w "$build_node" "cd $PWD && LD_LIBRARY_PATH=${HWLOC_INST}/lib:${LIBEV_INST}/lib:${PMIX_INST}/lib:${LD_LIBRARY_PATH} ./config.sh"
+        # shellcheck disable=SC2181
         if [ "$?" != "0" ]; then
             echo_error $LINENO "\"$item\" Configure error. Cannot continue."
             mv config.log config.log.bak
             exit 1
         fi
     fi
+
     if [ ! -f ".deploy_build_flag" ]; then
-        pdsh -S -w $build_node "cd $PWD && make -j $build_cpus"
+        pdsh -S -w "$build_node" "cd $PWD && make -j $build_cpus"
         ret=$?
         if [ "$ret" != "0" ]; then
             echo_error $LINENO "\"$item\" Build error. Cannot continue."
             exit 1
         fi
-        echo 1 > .deploy_build_flag
+        echo 1 >.deploy_build_flag
     fi
-    pdsh -S -w $build_node "cd $PWD && make -j $build_cpus install"
+
+    pdsh -S -w "$build_node" "cd $PWD && make -j $build_cpus install"
     ret=$?
+    # shellcheck disable=SC2181
+
     if [ "$?" != "0" ]; then
-        echo_error $LINENO "\"$item\" `make install` error. Cannot continue."
+        echo_error $LINENO "\"$item\" $(make install) error. Cannot continue."
         exit 1
     fi
 
-    if [ $item = "slurm" ]; then
-         pdsh -S -w $build_node "cd $PWD/contribs/pmi && make -j $build_cpus install"
-         pdsh -S -w $build_node "cd $PWD/contribs/pmi2 && make -j $build_cpus install"
+    if [ "$item" = "slurm" ]; then
+        pdsh -S -w "$build_node" "cd $PWD/contribs/pmi && make -j $build_cpus install"
+        pdsh -S -w "$build_node" "cd $PWD/contribs/pmi2 && make -j $build_cpus install"
+    fi
+
+    if [ "$item" = "ompi" ]; then
+        # TODO: tmp workaround
+        echo "INFO: $SRC_DIR/$item/prrte/src/include/prrte_config.h will be removed"
+        rm -f "$SRC_DIR/$item/prrte/src/include/prrte_config.h"
```

Comment:
Please remove this workaround. The issue with `prrte_config.h` was fixed with https://github.com/open-mpi/ompi/pull/7667.